diff --git a/src/Plugin/Mail/MandrillMail.php b/src/Plugin/Mail/MandrillMail.php
index 59f3a57..a11f5a7 100644
--- a/src/Plugin/Mail/MandrillMail.php
+++ b/src/Plugin/Mail/MandrillMail.php
@@ -117,8 +117,16 @@ class MandrillMail implements MailInterface {
     if (isset($message['params']['mandrill']['header'])) {
       $headers = $message['params']['mandrill']['header'] + $headers;
     }
-    if (!empty($message['from_email']) && empty($headers['Reply-To'])) {
-      $headers['Reply-To'] = $message['from_email'];
+    // different modules are using different capitalization in the header
+    // array keys i.e. Reply-to and Reply-To
+    $header_key_map=array();
+    foreach(array_keys($headers) as $header_key){
+      $header_key_map[strtolower($header_key)]=$header_key;
+    }
+    //if the header reply to is not set or empty addd the FROM address to as a Reply-to
+    if (!empty($message['from_email']) && (!isset($header_key_map['reply-to']) || empty($headers[$header_key_map['reply-to']]))) {
+      $reply_to_key = isset($header_key_map['reply-to']) ? $header_key_map['reply-to'] : 'Reply-to';
+      $headers[$reply_to_key] = $message['from_email'];
     }
     // Prepare attachments.
     $attachments = array();
